### PR TITLE
Modulr network schemes

### PR DESCRIPTION
**Pull request by TechiRexach**
**OUTDATED:**
We are not sure of making the differentiation by currency, but it is the simplest and safest way that we have found.
Any ideas?

-----------------------------------

**UPDATE:**
As Modulr, depending on the type of payment, it sends us the information in one way or another, the most consistent way to obtain the schema and the network is through the **type** attribute, that in some cases it is necessary to obtain from the associated transaction.

To make it more efficient, we have chosen to include the call to the transaction as optional, but for consistency in our API, a default true has been established to obtain it, because we believe that it is better to have more information than less.

### CODE REVIEW

File: lib/modulr/resources/payments/payment.rb [hunk 1]
```diff
@@ -17,6 +17,14 @@ class Payment < Base
...
           super(raw_response, attributes)
           @details = parse_details(attributes)
+          @network = parse_network(attributes[:details][:currency])
+        end
+
+        private def parse_network(currency)
+          currency = currency.upcase
+
+          return "SEPA" if currency == "EUR"
+          return "FPS" if currency == "GBP"
```
Comment (carloslopdev) [inline lib/modulr/resources/payments/payment.rb:27]: We are using `"UK_FPS"`, so maybe this string should be named the same way.

Check `app/domains/banking/services/processors/faster_payments/uk_fps.rb:9`

**Review by carloslopdev — APPROVED** [review]

**Comment by aitor** [comment]
I'm not very happy with this idea. It's at best ambiguous, as we could have payments in GBP that are no going through Faster Payments. 

I'd suggest two ways:
* Use the `schemeInfo` details to determine the network, eg. `SEPA_INSTANT` and `SEPA_CREDIT_TRANSFER` go to `SEPA`, anything starting with `MODULO` goes to FPS, anything else raises an exception. 
* Use the transaction generated by the payment, specifically the `type` attribute. `PO_SECT`, `PI_SECT`,   `PO_SEPA_INST` and `PI_SEPA_INST` go to `SEPA`,  `PO_FAST` and `PI_FAST` go to FPS, anything else raises an exception. 

This is an important consistency issue should I'd like to hear any other takes by @devengoapp/dev

**Comment by aitor** [comment]
Triggering test.

File: lib/modulr/api/payments_service.rb [hunk 2]
```diff
@@ -5,10 +5,18 @@ module API
     class PaymentsService < Service
       def find(id:)
         response = client.get("/payments", { id: id })
-        payment_attributes = response.body[:content]&.first
-        raise NotFoundError, "Payment #{id} not found" unless payment_attributes
+        @payment_attributes = response.body[:content]&.first
+        raise NotFoundError, "Payment #{id} not found" unless @payment_attributes
 
-        Resources::Payments::Payment.new(response.env[:raw_body], payment_attributes)
+        type =  if outgoing && !internal
```
Comment (Franx0) [inline lib/modulr/api/payments_service.rb:11]: I'm not sure about it 🤔 What happen with incoming payments?

Comment (Franx0) [reply lib/modulr/api/payments_service.rb:11]: Yes, but inside details, we should maintain at least same structure, isn't it? 🤔

File: spec/fixtures/payments/find/incoming/responses/success_faster_payments.http [hunk 3]
```diff
@@ -30,61 +30,61 @@ x-xss-protection: 1; mode=block
...
-        "payerName": "Aitor Garcia Rey",
+        "payerName": "John",
         "payer": {
-          "name": "Aitor Garcia Rey",
+          "name": "John",
           "identifier": {
             "type": "SCAN",
-            "accountNumber": "P12642236",
-            "sortCode": "TRWIBEBB",
+            "accountNumber": "P12642237",
```
Comment (Franx0) [inline spec/fixtures/payments/find/incoming/responses/success_faster_payments.http:47]: 🤔

File: lib/modulr/resources/payments/payment.rb [hunk 4]
```diff
@@ -51,18 +52,72 @@ def initialize(raw_response, attributes = {})
...
+          when "PI_SEPA_INST", "PO_SEPA_INST"
+            sepa_instant
+          when "PI_FAST", "PO_FAST"
+            faster_payments
+          when "ACCOUNT", "INT_INTERC"
+            internal
           else
-            Details::Outgoing::General.new(nil, details)
+            @network = nil
+            @scheme = nil
```
Comment (Franx0) [inline lib/modulr/resources/payments/payment.rb:86]: Should we raise an error here instead of null values? 🤔

Comment (Franx0) [reply lib/modulr/resources/payments/payment.rb:86]:
```suggestion
            raise "Unable to find network and scheme for payment with ID: #{id} and Type: #{type}"
```

**Pull request merged**